cranelift/codegen/meta/src/cdsl/typevar.rs

```diff
@@ -331,8 +311,6 @@ impl DerivedFunc {
             DerivedFunc::AsBool => "as_bool",
             DerivedFunc::HalfWidth => "half_width",
             DerivedFunc::DoubleWidth => "double_width",
-            DerivedFunc::HalfVector => "half_vector",
-            DerivedFunc::DoubleVector => "double_vector",
```

Comment:
Should we also remove them from the `Type` type?

Reply:
Looks like they are used in `fn image(&self, derived_func: DerivedFunc) -> TypeSet` on L#385. I'll admit, I'm not sure about the description of the `image` function or what it does

Reply:
It is fine to keep it then.

Reply:
I'm not 100% sure here, but I think you were right the first time, @bjorn3. Although `TypeSet::half_vector` and `TypeSet::double_vector` are still convenient for the implementation of `DerivedFunc::MergeLanes` and `DerivedFunc::SplitLanes`, the `Type::half_vector` method in `cranelift/codegen/src/ir/types.rs` doesn't seem to be used anywhere after this so I think it can be removed.

Reply:
Ah I see what bjorn meant now, makes sense. I'll just remove the method and inline its contents in  `Type::merge_lanes`